### PR TITLE
DOC/STY: docstrings in params, registry

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #860, #565 (partial)

Updates docstrings style and adds missing docstrings to 
- _params
- registration_test_class
- test_params
- test_registry

Also fixes a few style things from previous scrub pulls:
- import order
- "new style" declaration of classes

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?

via flake-docstrings:
```
flake8 --select=D,E,F,W --count --statistics
```

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.2
* flake8, flake-docstrings

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes (part of series already documented)

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
Remaining flags for elsewhere in the code:
```
4     D100 Missing docstring in public module
28    D101 Missing docstring in public class
70    D102 Missing docstring in public method
5     D107 Missing docstring in __init__
3     D204 1 blank line required after class docstring
10    D205 1 blank line required between summary line and description
2     D209 Multi-line docstring closing quotes should be on a separate line
52    D210 No whitespaces allowed surrounding docstring text
259   D400 First line should end with a period
86    D401 First line should be in imperative mood
1     D403 First word of the first line should be properly capitalized
520
```

File: pysat/_params.py [hunk 1]
```diff
@@ -219,10 +221,12 @@ def __str__(self, long_str=True):
         return out_str
 
     def __getitem__(self, item):
+        """Get item from Parameters."""
         return self.data[item]
 
     def __setitem__(self, key, value):
-        # Update current settings
+        """Update current settings in Parameters."""
+
```
Comment (jklenzing) [inline pysat/_params.py:229]: Does python ignore docstrings for __setitem__ like it does with the setter on properties?  If so, this info may need to be put in __getitem__.

File: pysat/_params.py [hunk 2]
```diff
@@ -286,9 +290,11 @@ def _set_data_dirs(self, path=None, store=True):
...
 
     def clear_and_restart(self):
-        """Clears all stored settings and sets pysat defaults
+        """Clear all stored settings and sets pysat defaults.
 
-        pysat parameters without a default value are set to []
+        Note
+        ----
+        pysat parameters without a default value are set to [].
         """
```
Comment (aburrell) [inline pysat/_params.py:298]:
```suggestion
        pysat parameters without a default value are set to [].

        """
```

File: pysat/tests/registration_test_class.py [hunk 3]
```diff
@@ -3,6 +3,12 @@
 # Full author list can be found in .zenodo.json file
 # DOI:10.5281/zenodo.1199703
 # ----------------------------------------------------------------------------
+"""Standardized class and functions to test registration for pysat libraries.
+
+Note
+----
+Not directly called by pytest, but imported as part of test_registry.py.
+"""
```
Comment (aburrell) [inline pysat/tests/registration_test_class.py:11]:
```suggestion
Not directly called by pytest, but imported as part of test_registry.py.

"""
```

File: pysat/tests/test_params.py [hunk 4]
```diff
@@ -167,24 +174,23 @@ def test_clear_and_restart(self):
...
+        return
 
 
 class TestCIonly(CICleanSetup):
     """Tests where we mess with local settings.
 
+    Note
+    ----
     These only run in CI environments to avoid breaking an end user's setup
     """
```
Comment (aburrell) [inline pysat/tests/test_params.py:190]:
```suggestion
    These only run in CI environments to avoid breaking an end user's setup

    """
```

**Review by aburrell — COMMENTED** [review]
Here and in other recent PR's you've forgotten to add the empty line for multi-line docstrings before the closing quotation marks.

**Comment by jklenzing** [comment]
> Here and in other recent PR's you've forgotten to add the empty line for multi-line docstrings before the closing quotation marks.

I'm going to poke into some other flake8 options to see if I can make this trigger an error.

**Comment by jklenzing** [comment]
> > Here and in other recent PR's you've forgotten to add the empty line for multi-line docstrings before the closing quotation marks.
> 
> I'm going to poke into some other flake8 options to see if I can make this trigger an error.

Following up from discussion in other issues, the `hacking` package adds more errors to flake8, including empty lines for multi-line docstrings.  Using that to scan as well.

File: pysat/_params.py [hunk 5]
```diff
@@ -155,7 +157,7 @@ def __init__(self, path=None, create_new=False):
         return
 
     def __repr__(self):
-        """String describing Parameters instantiation parameters
+        """Describe Parameters instantiation parameters.
```
Comment (aburrell) [inline pysat/_params.py:160]:
```suggestion
        """Describe the `Parameters` instantiation parameters.
```

File: pysat/_params.py [hunk 6]
```diff
@@ -168,7 +170,7 @@ def __repr__(self):
         return out_str
 
     def __str__(self, long_str=True):
-        """String describing Parameters instance, variables, and attributes
+        """Describe Parameters instance, variables, and attributes.
```
Comment (aburrell) [inline pysat/_params.py:173]:
```suggestion
        """Describe the `Parameters` instance, variables, and attributes.
```

File: pysat/_params.py (hunk 1 above)
Comment (aburrell) [reply pysat/_params.py:229]: Not sure 🦈

Comment (jklenzing) [reply pysat/_params.py:229]: @rstoneback, is the above construction correct?

**Review by aburrell — COMMENTED** [review]
Fine by me, but not approving because you need @rstoneback to sign off on the outstanding issue.

File: pysat/_params.py (hunk 1 above)
Comment (rstoneback) [reply pysat/_params.py:229]: The docstrings are correct. If we want them in the API docs though then the special members option needs to be invoked. I'm ok with it as is.

**Review by rstoneback — APPROVED** [review]

File: pysat/_params.py (hunk 1 above)
Comment (jklenzing) [reply pysat/_params.py:229]: testing out locally, adding the :special-members: in the api file works.  I can push that here so it's taken care of.

Comment (jklenzing) [reply pysat/_params.py:229]: or conversely, we can make it a new issue to add throughout docs where it makes sense.

Comment (rstoneback) [reply pysat/_params.py:229]: I'm leaning more towards adding it where it makes sense.

**Pull request merged**